godtossing/Pos/Binary/GodTossing/Core.hs

```diff
@@ -29,32 +32,34 @@ instance Bi Commitment where
         return Commitment {..}
 
 instance Bi CommitmentsMap where
-    put = put . toList
+    sizeNPut = putField (toList . getCommitmentsMap)
     get = label "CommitmentsMap" $ mkCommitmentsMap <$> get
 
 instance Bi VssCertificate where
```

Comment:
Why not implement it using TH then? What's the general policy on using TH-generated instances now?